src/io/test.zig

```diff
@@ -739,3 +739,105 @@ test "pipe data over socket" {
         }
     }.run();
 }
+
+test "cancel" {
+    const checksum = @import("../vsr/checksum.zig").checksum;
+    const allocator = std.testing.allocator;
+    const file_path = "test_cancel";
+    const read_count = 8;
+    const read_size = 1024 * 16;
+
+    // For this test to be useful, we rely on open(DIRECT).
+    // (See below).
+    if (builtin.target.os.tag != .linux) return;
+
+    try struct {
+        const Context = @This();
+
+        io: IO,
+        canceled: bool = false,
+
+        fn run_test() !void {
+            defer std.fs.cwd().deleteFile(file_path) catch {};
+
+            var context: Context = .{ .io = try IO.init(32, 0) };
+            defer context.io.deinit();
+
+            {
+                // Initialize the file and fill it with test data.
+                const file = try std.fs.cwd().createFile(file_path, .{
+                    .read = true,
```

Comment:
meant to be `.write = true`? And this could use `std.fs.cwd().writeFile` perhaps? 

As a rule of thumb, it's better to use bulk file operations if possible, where you don't create a file descriptor yourself and let the stdlib to handle  it for you.